TestFoundation/TestURL.swift

```diff
@@ -394,11 +394,20 @@ class TestURL : XCTestCase {
         }
         
         // tmp is special because it is symlinked to /private/tmp and this /private prefix should be dropped,
-        // so tmp is tmp. On Linux tmp is not symlinked so it would be the same.
+        // so tmp is tmp. On Linux tmp is not symlinked so it would be the same. In Android it doesn't exist
+        // so we have to use NSTemporaryDirectory to be safe.
+        #if !os(Android)
```

Comment:
This is going to be needed for Windows as well.  I think that this is safe to do always (as in always use `NSTemporaryDirectory`.  @millenomi - thoughts?

Reply:
I agree we should remove the hardcoded `/tmp` and just use `NSTemporaryDirectory` for all platforms. I think the test just want a directory that exists, it doesn't actually write to it anyway.

Reply:
In Darwin, at least, I got a `NSTemporaryDirectory` which was `/var/<letters and numbers>`. So the checks against resolving symlinks are going to be wrong if we use `NSTemporaryDirectory`.

A more resilient idea will be creating the directories ourselves, but it might also be complicated in Windows (there's no symlinks until recent versions of Windows, and you need special permissions). In the end `resolvingSymlinksInPath` does only a couple of things: removes empty components, removes `.`, resolves `..`, resolves symlinks, and removes the `/private` prefix. Most of the checks can be done against non-existing paths, except the ones that resolve symlinks (which can be done in any temporary directory), the rest should not access the FS.

If everyone agrees, I can do the changes, and I think it will work on Windows too.

Reply:
I have some PRs coming where `TestFileManager` will create its own directories.

I recommend creating your own directories, for instance: `NSTemporaryDirectory()` appending `./org.swift.Foundation.TestFoundation.\(pid)/TestURL/etc. etc.`

Reply:
@drodriguez See the test code in https://github.com/apple/swift-corelibs-foundation/pull/2149